tools/sz_repo_cli/lib/src/common/src/concurrent_package_task_runner.dart

```diff
@@ -0,0 +1,304 @@
+// Copyright (c) 2023 Sharezone UG (haftungsbeschränkt)
+// Licensed under the EUPL-1.2-or-later.
+//
+// You may obtain a copy of the Licence at:
+// https://joinup.ec.europa.eu/software/page/eupl
+//
+// SPDX-License-Identifier: EUPL-1.2
+
+import 'dart:async';
+import 'dart:collection';
+
+import 'package:meta/meta.dart';
+import 'package:optional/optional.dart';
+import 'package:rxdart/subjects.dart';
+import 'package:rxdart/transformers.dart';
+import 'package:sz_repo_cli/src/common/common.dart';
+
+class ConcurrentPackageTaskRunner {
+  DateTime Function() _getCurrentDateTime;
+
+  ConcurrentPackageTaskRunner({
+    DateTime Function() getCurrentDateTime,
+  }) {
+    _getCurrentDateTime = getCurrentDateTime ?? () => DateTime.now();
+  }
+
+  /// Run [runTask] for every package from the [packageStream].
+  Stream<PackageTask> runTaskForPackages({
+    /// Stream of packages for which [runTask] will be called.
+    ///
+    /// Most often this will be [SharezoneRepo.streamPackages], maybe with a
+    /// filter (e.g. `.where((package) => package.hasTestDirectory`) or
+    /// something similar.
+    @required Stream<Package> packageStream,
+    @required Future<void> Function(Package package) runTask,
+
+    /// If 0, null or negative then every package will be processed at once.
+    /// Else [maxNumberOfPackagesBeingProcessedConcurrently] are processed
+    /// concurrently,
```

Comment:
```suggestion
    /// Defines the number of packages that being processed concurrently.
    ///
    /// If 0, null or negative then every package will be processed at once.
```

Even when the first sentence repeats the name of the variable, is this maybe a better documentation that follows Effective Dart in the following points:
* [DO start doc comments with a single-sentence summary.](https://dart.dev/guides/language/effective-dart/documentation#do-start-doc-comments-with-a-single-sentence-summary)
* [DO separate the first sentence of a doc comment into its own paragraph.](https://dart.dev/guides/language/effective-dart/documentation#do-separate-the-first-sentence-of-a-doc-comment-into-its-own-paragraph)

Reply:
Thanks!